### PR TITLE
Fix: Added packages needed for "open with" in dolphin

### PR DESCRIPTION
**Pull request by boudywho**
# Pull Request

## Description

- Just added `kservice5 kdbusaddons5 kfilemetadata5 kconfig5 kcoreaddons5` to fix "open with" menu in dolphin

- I had an issue where I couldn't see any apps in "open with" and files couldn't be opened with vscode (which is the default app), other file managers could open it with no issues.

- Installing those packages fixed the problem.

## Type of change

Please put an `x` in the boxes that apply:

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] **Documentation update** (non-breaking change; modified files are limited to the documentations)
- [ ] **Technical debt** (a code change that does not fix a bug or add a feature but makes something clearer for devs)
- [ ] **Other** (provide details below)

## Checklist

Please put an `x` in the boxes that apply:

- [X] I have read the [CONTRIBUTING](https://github.com/prasanthrangan/hyprdots/blob/main/CONTRIBUTING.md) document.
- [X] My code follows the code style of this project.
- [X] My commit message follows the [commit guidelines](https://github.com/prasanthrangan/hyprdots/blob/main/CONTRIBUTING.md#git-commit-messages).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added a changelog entry.
- [X] I have added necessary comments/documentation to my code.
- [ ] I have added tests to cover my changes.
- [X] I have tested my code locally and it works as expected.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Scripts/custom_hypr.lst [hunk 1]
```diff
@@ -49,6 +49,18 @@ imagemagick                                            # for image processing
...
+kfilemetadata5                                          # for "open with" menu in dolphin
+kconfig5                                                # for "open with" menu in dolphin
+kcoreaddons5                                           # for "open with" menu in dolphin
+kcrash5                                                # for "open with" menu in dolphin
+kguiaddons5                                            # for "open with" menu in dolphin
+ki18n5                                                 # for "open with" menu in dolphin
+kitemviews5                                            # for "open with" menu in dolphin
+kwidgetsaddons5                                        # for "open with" menu in dolphin
+kwindowsystem5                                         # for "open with" menu in dolphin
+kservice5                                              # for "open with" menu in dolphin
```
Comment (kRHYME7) [inline Scripts/custom_hypr.lst:63]: dupe?

Comment (boudywho) [reply Scripts/custom_hypr.lst]: > dupe? L52

my bad, fixed.

**Review by akashgagda — COMMENTED** [review]
It's fix my issue also

**Comment by boudywho** [comment]
> Do we really need all of this?
> 
> BTW I installed it all directly then Everything works as expected. Thanks
> 
> I don't actually understand Why my other machine don't have this issue.

Glad it helped. 

You don't need all of them, in fact, I removed some on my machine and it worked fine, but while testing with other people giving them the exact packages, it didn't work for some reason.

Do you have KDE installed on your other machine? (or had) because on my other machine, I didn't have this issue because I already had KDE on another profile.

**Comment by kRHYME7** [comment]
>Do you have KDE installed on your other machine? (or had) because on my other machine, I didn't have this issue because I already had KDE on another profile.

Nope, My other machine  is a fresh arch install + Hyprdots.

Surprisingly all packages are present. 
I wrap the list on a script and check what pkg depends on them.  Maybe we could narrow this packages 

Maybe this data is useful 

```

Name            : kservice5
Depends On      : ki18n5  kconfig5  kdbusaddons5  kcoreaddons5
Required By     : kded5  kwallet5
Install Reason  : Installed as a dependency for another package
*******************************************************************
Name            : kdbusaddons5
Depends On      : qt5-x11extras
Required By     : kservice5
Install Reason  : Installed as a dependency for another package
*******************************************************************
Name            : kfilemetadata5
Depends On      : ki18n5  karchive5  kconfig5  kcoreaddons5  exiv2  poppler-qt5  taglib  ffmpeg  ebook-tools
Required By     : None
Install Reason  : Installed as a dependency for another package
*******************************************************************
Name            : kconfig5
Depends On      : qt5-base
Required By     : kactivities5  kcompletion5  kconfigwidgets5  kdesu5  kfilemetadata5  kglobalaccel5  knotifications5  kservice5
Install Reason  : Installed as a dependency for another package
*******************************************************************
Name            : kcoreaddons5
Depends On      : qt5-base
Required By     : kactivities5  kauth5  kcrash5  kfilemetadata5  kglobalaccel5  kjobwidgets5  knotifications5  kpackage5  kpty5  kservice5
Install Reason  : Installed as a dependency for another package
*******************************************************************
Name            : kcrash5
Depends On      : kcoreaddons5  kwindowsystem5
Required By     : kded5
Install Reason  : Installed as a dependency for another package
*******************************************************************
Name            : kguiaddons5
Depends On      : qt5-x11extras  qt5-wayland
Required By     : kconfigwidgets5
Install Reason  : Installed as a dependency for another package
*******************************************************************
Name            : ki18n5
Depends On      : qt5-declarative  iso-codes
Required By     : kconfigwidgets5  kfilemetadata5  kpackage5  kpty5  kservice5
Install Reason  : Installed as a dependency for another package
*******************************************************************
Name            : kitemviews5
Depends On      : qt5-base
Required By     : kxmlgui5
Install Reason  : Installed as a dependency for another package
*******************************************************************
Name            : kwidgetsaddons5
Depends On      : qt5-base
Required By     : kcompletion5  kconfigwidgets5  kjobwidgets5
Install Reason  : Installed as a dependency for another package
*******************************************************************
Name            : kwindowsystem5
Depends On      : qt5-x11extras
Required By     : kcrash5  kglobalaccel5  knotifications5
Install Reason  : Installed as a dependency for another package
*******************************************************************
Name            : kservice5
Depends On      : ki18n5  kconfig5  kdbusaddons5  kcoreaddons5
Required By     : kded5  kwallet5
Install Reason  : Installed as a dependency for another package
*******************************************************************
```

**Comment by kRHYME7** [comment]
For a glimpse all of the other packages needs this packages: 

```
kservice5
qt5-base

```

**Comment by kRHYME7** [comment]
Look kservices can bring some of the packages as they  are required by kservices

**Comment by boudywho** [comment]
> > Do you have KDE installed on your other machine? (or had) because on my other machine, I didn't have this issue because I already had KDE on another profile.
> 
> Nope, My other machine is a fresh arch install + Hyprdots.
> 
> Surprisingly all packages are present. I wrap the list on a script and check what pkg depends on them. Maybe we could narrow this packages
> 
> Maybe this data is useful
> 
> ```
> 
> Name            : kservice5
> Depends On      : ki18n5  kconfig5  kdbusaddons5  kcoreaddons5
> Required By     : kded5  kwallet5
> Install Reason  : Installed as a dependency for another package
> *******************************************************************
> Name            : kdbusaddons5
> Depends On      : qt5-x11extras
> Required By     : kservice5
> Install Reason  : Installed as a dependency for another package
> *******************************************************************
> Name            : kfilemetadata5
> Depends On      : ki18n5  karchive5  kconfig5  kcoreaddons5  exiv2  poppler-qt5  taglib  ffmpeg  ebook-tools
> Required By     : None
> Install Reason  : Installed as a dependency for another package
> *******************************************************************
> Name            : kconfig5
> Depends On      : qt5-base
> Required By     : kactivities5  kcompletion5  kconfigwidgets5  kdesu5  kfilemetadata5  kglobalaccel5  knotifications5  kservice5
> Install Reason  : Installed as a dependency for another package
> *******************************************************************
> Name            : kcoreaddons5
> Depends On      : qt5-base
> Required By     : kactivities5  kauth5  kcrash5  kfilemetadata5  kglobalaccel5  kjobwidgets5  knotifications5  kpackage5  kpty5  kservice5
> Install Reason  : Installed as a dependency for another package
> *******************************************************************
> Name            : kcrash5
> Depends On      : kcoreaddons5  kwindowsystem5
> Required By     : kded5
> Install Reason  : Installed as a dependency for another package
> *******************************************************************
> Name            : kguiaddons5
> Depends On      : qt5-x11extras  qt5-wayland
> Required By     : kconfigwidgets5
> Install Reason  : Installed as a dependency for another package
> *******************************************************************
> Name            : ki18n5
> Depends On      : qt5-declarative  iso-codes
> Required By     : kconfigwidgets5  kfilemetadata5  kpackage5  kpty5  kservice5
> Install Reason  : Installed as a dependency for another package
> *******************************************************************
> Name            : kitemviews5
> Depends On      : qt5-base
> Required By     : kxmlgui5
> Install Reason  : Installed as a dependency for another package
> *******************************************************************
> Name            : kwidgetsaddons5
> Depends On      : qt5-base
> Required By     : kcompletion5  kconfigwidgets5  kjobwidgets5
> Install Reason  : Installed as a dependency for another package
> *******************************************************************
> Name            : kwindowsystem5
> Depends On      : qt5-x11extras
> Required By     : kcrash5  kglobalaccel5  knotifications5
> Install Reason  : Installed as a dependency for another package
> *******************************************************************
> Name            : kservice5
> Depends On      : ki18n5  kconfig5  kdbusaddons5  kcoreaddons5
> Required By     : kded5  kwallet5
> Install Reason  : Installed as a dependency for another package
> *******************************************************************
> ```

@kRHYME7  

- I uninstalled all these packages, but the "Open With" menu persisted. Interestingly, after uninstalling and then installing Vim (or any app that appears in the "open with" menu like VS Code), the menu disappeared. This suggests one (or a couple) of the removed packages might have a one-time effect on the menu.

- Narrowing it down, I focused on these packages: `kservice5, kdbusaddons5, kfilemetadata5, kconfig5, kcoreaddons5`. Try them out for yourself, and left me know if you could narrow it even more.

**Comment by kRHYME7** [comment]
``` kdbusaddons5, kfilemetadata5, kconfig5, kcoreaddons5 ``` 

These packages are required by ``` kservice5 ```, meaning they will be installed if we install ``` kservices5 ```.

**Comment by prasanthrangan** [comment]
dolphin has updated the dependency list based on qt6... so the latest version should work without these qt5 packages.

**Comment by kRHYME7** [comment]
> dolphin has updated the dependency list based on qt6... so the latest version should work without these qt5 packages.

I somewhat agree with this. As this happens only on 1 of my device. Maybe it's just pacman not handling it correctly during update (and my crazy stuff).

**Comment by boudywho** [comment]
> dolphin has updated the dependency list based on qt6... so the latest version should work without these qt5 packages.

Interesting. While it's true that Dolphin has updated its dependencies, I'm experiencing a different situation. When I remove the following packages, the 'open with' functionality in Dolphin breaks for me (on the latest version):

   - kservice5
   - kdbusaddons5
   - kfilemetadata5
   - kconfig5
   - kcoreaddons5

Could there be something specific to my system configuration that's still relying on these older packages? Or are there alternative solutions to ensure the 'open with' menu works without them?

**Comment by kRHYME7** [comment]
On my machine, Some of my apps rely on sub dependencies pointing to the pakages mentioned above. Example vlc relies on some qt5 deps so I can's remove those packages. It seems like pacman failed some compatibility update.

**Comment by prasanthrangan** [comment]
I think `kservice5` covers most of these dependencies -

image

so can you guys try installing `kservice5` and check if it fix the menu.

**Comment by boudywho** [comment]
> I think `kservice5` covers most of these dependencies -
> 
> image
> 
> so can you guys try installing `kservice5` and check if it fix the menu.

Installing just kservice5 resolved the issue. It doesn't work automatically, so I reinstalled vim, then it worked.

**Comment by ALEX5402** [comment]
> * `kservice5 kdbusaddons5 kfilemetadata5 kconfig5 kcoreaddons5`
thanks it fixed my issues with dolphin

**Review by kRHYME7 — APPROVED** [review]
Tried  ``` kservice5 ``` on fresh install fixes the issue . Can be merge.

**Review by prasanthrangan — APPROVED** [review]
great, thank you!

**Pull request merged**